lib/openshift/o_auth.rb

```diff
@@ -0,0 +1,28 @@
+module BushSlicer
+  class OAuth < ClusterResource
+    RESOURCE = "oauth"
+    # pls refer https://github.com/openshift-qe/output_refrences/tree/master/oauth for output example
+
+    # return a list of Hash with the user defined name mapping to the system generated name
```

Comment:
try to follow yardoc spec with `@return`. Can we return just a `Hash` where key is identity provider name and value is secret name?

Reply:
then you will not need `generated_htpasswd_name` method. You can just do `htpasswds(...)["idp_name"]` to get secret name (if provider exists).

Reply:
@akostadinov the use case from @yapei is that we need to find the generated name first before calling the `secret` object

Reply:
I mean instead of current example return value to return:
```
{
  "flexy-htpasswd-provider"=>"htpass-secret",
  "htpasswd"=>"htpasswd-2nhtj",
  "htpasswd-foo"=>"htpasswd-rtc5q",
  "htpasswd-bar"=>"htpasswd-8zlqs",
  "foobar"=>"htpasswd-txc9w"
}
```
I see no benefit in returning an array of hashes regardless of the use case. Or can you give an example where the other data structure provides benefits?

Reply:
@akostadinov, thanks for the clarification.  I've updated the PR.  PTAL.  Thanks!